### PR TITLE
replica_rac2: rm unused waitingForAdmissionState

### PR DESCRIPTION
**Pull request by pav-kv**
Also copy the comment explaining why we chose to track individual log indices rather then compressing them into a single "waiting" interval.

Part of #129508

### CODE REVIEW

File: pkg/kv/kvserver/kvflowcontrol/replica_rac2/admission.go [hunk 1]
```diff
@@ -192,116 +188,3 @@ func (p *lowPriOverrideState) getEffectivePriority(
...
-
-// waitingForAdmissionState records the indices of individual entries that are
-// waiting for admission in the AC queues. These are added after emerging as
-// MsgStorageAppend in handleRaftReady, hence we can expect monotonicity of
-// indices within a leader term, and if the term changes, we can assume that
-// non-monotonicity means raft log indices being overwritten and the log being
-// truncated. These assumptions simplify the processing of add, since it can
-// wipe out higher indices.
-//
-// We've chosen to track individual indices instead of the latest index, to
```
Comment (pav-kv) [inline pkg/kv/kvserver/kvflowcontrol/replica_rac2/admission.go:204]: Might want to copy some of this comment into `LogTracker`.

Comment (sumeerbhola) [reply pkg/kv/kvserver/kvflowcontrol/replica_rac2/admission.go:204]: > Might want to copy some of this comment into LogTracker.

Do you want to do this now, before we forget?

**Review by sumeerbhola — APPROVED** [review]
:lgtm: 

> Reviewed 4 of 4 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/130382#-:-O6RzS2n89e5XheNCCFK:bg7igkv)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @pav-kv)

**Comment by pav-kv** [comment]
bors r=sumeerbhola

**Comment by pav-kv** [comment]
bors r=sumeerbhola

**Pull request merged**